Beam/Include/Beam/Queues/FilteredPublisher.hpp

```diff
@@ -28,15 +29,15 @@ namespace Beam {
       public Details::GetPublisherType<PublisherType>::type,
       public QueueWriter<typename PublisherType::Type> {
     public:
-      typedef typename PublisherType::Type Type;
-      typedef typename Details::GetSnapshotType<PublisherType>::type Snapshot;
+      using Type = typename PublisherType::Type;
+      using Snapshot = typename Details::GetSnapshotType<PublisherType>::type;
 
       //! Defines the function used to filter values.
       /*!
         \param value The value to test.
         \return <code>true</code> iff the <i>value</i> is to be published.
       */
-      typedef std::function<bool (const Type& value)> FilterFunction;
+      using FilterFunction = std::function<bool(const Type&)>;
```

Comment:
The name of the parameter is missing, we use it because it aligns with the documentation.